### PR TITLE
fix!: improve training stability by adding the option to disable bias terms and add better profiling

### PR DESCRIPTION
**Pull request by MauritsBleeker**
This pull request introduces two main sets of changes: (1) improved configurability for bias usage in linear and MLP layers across multiple model and config classes, and (2) enhanced training stability monitoring, including more detailed gradient diagnostics and a new callback for logging training stability metrics. These updates make model architecture and training diagnostics more flexible and informative.

### CODE REVIEW

File: src/noether/core/callbacks/default/training_stability.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import math
+from typing import Any
+
+import torch.nn as nn
+
+from noether.core.callbacks.periodic import PeriodicCallback
+from noether.core.utils.model import compute_model_norm
+
+
+class TrainingStabilityCallback(PeriodicCallback):
```
Comment (HennerM) [inline src/noether/core/callbacks/default/training_stability.py]: I think something like `TrainingDyanmics` `TrainingInternals` might be a better name for this.

File: src/noether/core/callbacks/default/training_stability.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            return 0.0
+
+        total = 0
+        near_zero = 0
+        for p in model.parameters():
+            if p.grad is None:
+                continue
+            total += 1
+            if p.grad.norm() < eps:
+                near_zero += 1
```
Comment (HennerM) [inline src/noether/core/callbacks/default/training_stability.py]: this does count gradient tensors instead of the total number of gradients?

How about 

```suggestion
            total += p.numel()
            near_zero += (p.grad < eps).sum()
```

File: src/noether/core/callbacks/default/training_stability.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        for cur_name, cur_model in self.model.get_named_models().items():
+            optimizer = cur_model.optimizer
+            if optimizer is None or optimizer.last_grad_norm is None:
+                continue
+            norm = optimizer.last_grad_norm.item()
+            if math.isfinite(norm):
+                self.writer.add_scalar(f"profiling/optim/grad_norm/{cur_name}", norm)
+        model_norm = compute_model_norm(self.model).item()
+        self.writer.add_scalar("profiling/optim/model_norm", model_norm)
```
Comment (HennerM) [inline src/noether/core/callbacks/default/training_stability.py]: usually the term `profiling` is associated with performance profiling, not sure it's the best term

File: src/noether/core/callbacks/default/training_stability.py (hunk 1 above)
Comment (HennerM) [reply src/noether/core/callbacks/default/training_stability.py]: actually TrainingDiagnostics as used below sounds quite good

**Review by HennerM — APPROVED** [review]
nice, this is going to be very useful I believe!

File: src/noether/core/callbacks/default/training_stability.py (hunk 3 above)
Comment (Ndles) [reply src/noether/core/callbacks/default/training_stability.py]: alongside the comment before - `diagnostics` or `monitoring` would do, IMO

**Review by Ndles — APPROVED** [review]
good to go, unless you want to rename "profiling" directory

**Comment by Ndles** [comment]
also a note, for the PR Validation action to be happy, in case of a breaking change defined like this: `!fix: improve` you have to put "!" after the word "fix": `fix!: improve`.
